litellm/llms/snowflake/chat/transformation.py

```diff
@@ -0,0 +1,217 @@
+'''
+Support for Snowflake REST API 
+'''
+import httpx
+from typing import List, Optional, Tuple, Any, TYPE_CHECKING
+
+from litellm.types.llms.openai import AllMessageValues
+from litellm.utils import get_secret
+from litellm.types.utils import ModelResponse
+from litellm.types.llms.openai import ChatCompletionAssistantMessage
+from litellm.llms.databricks.streaming_utils import ModelResponseIterator
+
+from ...openai_like.chat.transformation import OpenAIGPTConfig
+
+if TYPE_CHECKING:
+    from litellm.litellm_core_utils.litellm_logging import Logging as _LiteLLMLoggingObj
+
+    LiteLLMLoggingObj = _LiteLLMLoggingObj
+else:
+    LiteLLMLoggingObj = Any
+
+class SnowflakeConfig(OpenAIGPTConfig):
+    """
+    source: https://docs.snowflake.com/en/sql-reference/functions/complete-snowflake-cortex
+    """
+
+    @classmethod
+    def get_config(cls):
+        return super().get_config()
+    
+    def get_supported_openai_params(self, model: str) -> List:
+        return [
+            "temperature",
+            "max_tokens",
+            "top_p",
+            "response_format"
+        ]
+            
+    def map_openai_params(
+        self,
+        non_default_params: dict,
+        optional_params: dict,
+        model: str,
+        drop_params: bool,
+    ) -> dict:
+        """
+        If any supported_openai_params are in non_default_params, add them to optional_params, so they are used in API call
+
+        Args:
+            non_default_params (dict): Non-default parameters to filter.
+            optional_params (dict): Optional parameters to update.
+            model (str): Model name for parameter support check.
+
+        Returns:
+            dict: Updated optional_params with supported non-default parameters.
+        """
+        supported_openai_params = self.get_supported_openai_params(model)
+        for param, value in non_default_params.items():
+            if param in supported_openai_params:
+                optional_params[param] = value
+        return optional_params
+    
+    def _convert_tool_response_to_message(
```

Comment:
not needed imo, snowflake does not support tool calling: https://docs.snowflake.com/en/sql-reference/functions/complete-snowflake-cortex